### PR TITLE
IDE-763 additional right click copy/paste button

### PR DESCRIPTION
**Pull request by dehilsterlexis**

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith to look at the keystroke accelerators

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: eclide/EclDlgBuilder.cpp [hunk 1]
```diff
@@ -40,6 +40,13 @@ void CBuilderDlg::GetTitle(CString & title)
     title = (m_view.IsDirty() ? _T("*") : _T("")) + name;
 }
 
+void CBuilderDlg::PathToClipboard() {
+    std::_tstring path = m_attribute->GetQualifiedLabel(false, true);
```
Comment (GordonSmith) [inline eclide/EclDlgBuilder.cpp:44]: I thought the m_attribute had a getPath function (when its an IFileAttribute)?

File: eclide/RepositoryFrame.h [hunk 2]
```diff
@@ -86,7 +89,8 @@ class CRepositoryFrameT :	public CRepositorySlotImpl,
         m.EnableMenuItem(ID_REPOSITORY_OPEN, state.CanOpen ? MF_ENABLED : MF_GRAYED);
         m.EnableMenuItem(ID_REPOSITORY_OPENBUILDER, state.CanOpenBuilder ? MF_ENABLED : MF_GRAYED);
         m.EnableMenuItem(ID_EDIT_LOCATEFILEINEXPLORER, state.CanOpenInExplorer ? MF_ENABLED : MF_GRAYED);
-        m.EnableMenuItem(ID_REPOSITORY_COPY, state.CanCopy ? MF_ENABLED : MF_GRAYED);
+        m.EnableMenuItem(ID_REPOSITORY_COPY, MF_ENABLED);
```
Comment (GordonSmith) [inline eclide/RepositoryFrame.h:92]: Isn't `state.CanCopy` still needed?

File: eclide/RepositoryFrame.h [hunk 3]
```diff
@@ -127,6 +131,12 @@ class CRepositoryFrameT :	public CRepositorySlotImpl,
             copyStr.Format(_T("&Copy \"%s\""), s.attrs.begin()->get()->GetQualifiedLabel(true));
             m.ModifyMenu(ID_REPOSITORY_COPY, MF_BYCOMMAND | MF_STRING, ID_REPOSITORY_COPY, copyStr); 
 
+            m_copyPathStr = s.attrs.begin()->get()->GetQualifiedLabel(true, true);
```
Comment (GordonSmith) [inline eclide/RepositoryFrame.h:134]: Ditto - check if its an File Attribute

File: eclide/RepositoryFrame.h [hunk 4]
```diff
@@ -288,6 +298,12 @@ class CRepositoryFrameT :	public CRepositorySlotImpl,
                 }
             }
             break;
+        case ID_REPOSITORY_COPYPATH:
+            ATLASSERT(s.attrs.size() > 0);
+            {
+                ::SetClipboard(m_copyPathStr);
```
Comment (GordonSmith) [inline eclide/RepositoryFrame.h:304]: Bleh (but probably ok to leave as is).

File: eclide/EclDlgBuilder.cpp (hunk 1 above)
Comment (dehilsterlexis) [reply eclide/EclDlgBuilder.cpp:45]: @GordonSmith getPath gets the entire path including the drive and repository location. The request was for the relative path.

File: eclide/RepositoryFrame.h (hunk 2 above)
Comment (dehilsterlexis) [reply eclide/RepositoryFrame.h:92]: @GordonSmith you are correct. Correcting.

File: eclide/EclDlg.h [hunk 5]
```diff
@@ -118,9 +120,25 @@ class CEclDlgQBImpl:
         CString copyStr, titleStr;
         GetTitle(titleStr);
         titleStr.TrimLeft(_T("*"));
+
         copyStr.Format(_T("&Copy \"%s\"\tCtrl+Alt+C"), titleStr);
         m.ModifyMenu(ID_REPOSITORY_COPY, MF_BYCOMMAND | MF_STRING, ID_REPOSITORY_COPY, copyStr);
+
+        IAttribute *attr = m_view.GetAttribute();
+        if (attr) {
```
Comment (GordonSmith) [inline eclide/EclDlg.h:128]: Test if attr is an instance of "IDiskAttribute" and if true use the "GetPath" function:
```
    if (CComQIPtr<IDiskAttribute> diskAttr = attr) {
        ... diskAttr->GetPath()); ...
    }
```

Comment (GordonSmith) [reply eclide/EclDlg.h:128]: You can use the same test to disable the copy path option when not a DiskAttr?

File: eclide/EclDlgBuilder.cpp (hunk 1 above)
Comment (GordonSmith) [reply eclide/EclDlgBuilder.cpp:45]: You can add Relative Path to the DiskAttribute class?  (would save duplicating this code).

Comment (dehilsterlexis) [reply eclide/EclDlgBuilder.cpp:45]: @GordonSmith can do. Good idea.

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Pull request merged**